### PR TITLE
CAD-608: Structured logging in tx generator.

### PR DESCRIPTION
**Pull request by denisshevchenko**
Move tx generator to the structured logging.

### CODE REVIEW

File: cardano-node/src/Cardano/CLI/Benchmarking/Tx/NodeToNode.hs [hunk 1]
```diff
@@ -46,8 +55,45 @@ type SendRecvConnect = WithMuxBearer
...
+
+--------------------------------------------------------------------------------------
+
 type SendRecvTxSubmission blk = TraceSendRecv (TxSubmission (GenTxId blk) (GenTx blk))
 
+instance forall blk . (RunNode blk) => ToObject (SendRecvTxSubmission blk) where
+  toObject MinimalVerbosity _ = emptyObject -- do not log
+  toObject NormalVerbosity _  = mkObject ["kind" .= String "SendRecvTxSubmission"]
+  toObject MaximalVerbosity _ =
+    mkObject [ "kind"   .= String "SendRecvTxSubmission"
```
Comment (CodiePP) [inline cardano-node/src/Cardano/CLI/Benchmarking/Tx/NodeToNode.hs]: this is the same as "NormalVerbosity"
intended?

File: cardano-node/src/Cardano/CLI/Benchmarking/Tx/NodeToNode.hs [hunk 2]
```diff
@@ -46,8 +55,45 @@ type SendRecvConnect = WithMuxBearer
...
+             ]
+
+instance forall blk . (RunNode blk) => DefineSeverity (SendRecvTxSubmission blk)
+
```
Comment (CodiePP) [inline cardano-node/src/Cardano/CLI/Benchmarking/Tx/NodeToNode.hs]: the default is "Debug"
intended?

File: cardano-node/src/Cardano/CLI/Tx.hs [hunk 3]
```diff
@@ -34,7 +34,7 @@ import           Data.Text (Text)
 import qualified Data.Text as T
 import           Formatting ((%), sformat)
 
-import           Control.Tracer (traceWith, stdoutTracer)
+import           Control.Tracer (traceWith, nullTracer, stdoutTracer)
```
Comment (CodiePP) [inline cardano-node/src/Cardano/CLI/Tx.hs]: do you still use 'stdoutTracer'?

**Comment by CodiePP** [comment]
output from generator.json:
```
{"at":"2020-02-17T17:10:15.39Z","env":"1.5.0:19cd0","ns":["cardano","cli","generate-txs","benchmark"],"data":{"kind":"TraceBenchTxSubStart","txIds":["<txId>","<txId>"]},"app":[],"msg":"","pid":"17763","loc":null,"host":"","sev":"Debug","thread":"141"}
{"at":"2020-02-17T17:10:15.39Z","env":"1.5.0:19cd0","ns":["cardano","cli","generate-txs","benchmark"],"data":{"kind":"TraceBenchTxSubServAck","txIds":["<txId>","<txId>"]},"app":[],"msg":"","pid":"17763","loc":null,"host":"","sev":"Debug","thread":"141"}
{"at":"2020-02-17T17:10:15.39Z","env":"1.5.0:19cd0","ns":["cardano","cli","generate-txs","submit"],"data":{"kind":"SendRecvTxSubmission"},"app":[],"msg":"","pid":"17763","loc":null,"host":"","sev":"Debug","thread":"148"}
{"at":"2020-02-17T17:10:15.39Z","env":"1.5.0:19cd0","ns":["cardano","cli","generate-txs","submit"],"data":{"kind":"SendRecvTxSubmission"},"app":[],"msg":"","pid":"17763","loc":null,"host":"","sev":"Debug","thread":"148"}
{"at":"2020-02-17T17:10:15.39Z","env":"1.5.0:19cd0","ns":["cardano","cli","generate-txs","benchmark"],"data":{"kind":"TraceBenchTxSubServReq","txIds":["<txId>"]},"app":[],"msg":"","pid":"17763","loc":null,"host":"","sev":"Debug","thread":"141"}
{"at":"2020-02-17T17:10:15.39Z","env":"1.5.0:19cd0","ns":["cardano","cli","generate-txs","submit"],"data":{"kind":"SendRecvTxSubmission"},"app":[],"msg":"","pid":"17763","loc":null,"host":"","sev":"Debug","thread":"148"}
{"at":"2020-02-17T17:10:15.39Z","env":"1.5.0:19cd0","ns":["cardano","cli","generate-txs","submit"],"data":{"kind":"SendRecvTxSubmission"},"app":[],"msg":"","pid":"17763","loc":null,"host":"","sev":"Debug","thread":"148"}
{"at":"2020-02-17T17:10:15.39Z","env":"1.5.0:19cd0","ns":["cardano","cli","generate-txs","benchmark"],"data":{"kind":"TraceBenchTxSubServReq","txIds":["<txId>","<txId>"]},"app":[],"msg":"","pid":"17763","loc":null,"host":"","sev":"Debug","thread":"141"}
{"at":"2020-02-17T17:10:15.39Z","env":"1.5.0:19cd0","ns":["cardano","cli","generate-txs","submit"],"data":{"kind":"SendRecvTxSubmission"},"app":[],"msg":"","pid":"17763","loc":null,"host":"","sev":"Debug","thread":"148"}
{"at":"2020-02-17T17:10:15.39Z","env":"1.5.0:19cd0","ns":["cardano","cli","generate-txs","submit"],"data":{"kind":"SendRecvTxSubmission"},"app":[],"msg":"","pid":"17763","loc":null,"host":"","sev":"Debug","thread":"148"}
{"at":"2020-02-17T17:10:15.39Z","env":"1.5.0:19cd0","ns":["cardano","cli","generate-txs","benchmark"],"data":{"kind":"TraceBenchTxSubRateLimit","limit":0.194734651},"app":[],"msg":"","pid":"17763","loc":null,"host":"","sev":"Debug","thread":"141"}
```

File: cardano-node/src/Cardano/CLI/Tx.hs (hunk 3 above)
Comment (denisshevchenko) [reply cardano-node/src/Cardano/CLI/Tx.hs]: There're still some places in this module where `stdoutTracer` is used:

1. `issueGenesisUTxOExpenditure`
2. `issueUTxOExpenditure`
3. `nodeSubmitTx`

File: cardano-node/src/Cardano/CLI/Benchmarking/Tx/NodeToNode.hs [hunk 4]
```diff
@@ -46,8 +57,47 @@ type SendRecvConnect = WithMuxBearer
...
+--------------------------------------------------------------------------------------
+
 type SendRecvTxSubmission blk = TraceSendRecv (TxSubmission (GenTxId blk) (GenTx blk))
 
+instance forall blk . (RunNode blk) => ToObject (SendRecvTxSubmission blk) where
+  toObject MinimalVerbosity _ = emptyObject -- do not log
+  toObject NormalVerbosity _ =
+    mkObject ["kind" .= String "SendRecvTxSubmission"]
+  toObject MaximalVerbosity _ =
+    mkObject [ "kind"   .= String "SendRecvTxSubmission"
```
Comment (CodiePP) [inline cardano-node/src/Cardano/CLI/Benchmarking/Tx/NodeToNode.hs]: we need more info:
* timestamp
* tx id

Comment (denisshevchenko) [reply cardano-node/src/Cardano/CLI/Benchmarking/Tx/NodeToNode.hs]: Some examples:

```
{"at":"2020-02-19T09:11:18.59Z","env":"1.5.0:baf2a","ns":["cardano","cli","generate-txs","submit"],"data":{"ackNum":0,"time":"2020-02-19 09:11:18.598503221 UTC","reqNum":3,"kind":"TxSubmissionRecvRequestTxIds"},"app":[],"msg":"","pid":"1146","loc":null,"host":"","sev":"Debug","thread":"96"}
```
```
{"at":"2020-02-19T09:11:18.63Z","env":"1.5.0:baf2a","ns":["cardano","cli","generate-txs","submit"],"data":{"time":"2020-02-19 09:11:18.630378862 UTC","kind":"TxSubmissionSendNBReplyTxIds","txIds":[["txid: txid:7252f9bb",307],["txid: txid:f680bb77",307],["txid: txid:ff2a85b8",307]]},"app":[],"msg":"","pid":"1146","loc":null,"host":"","sev":"Debug","thread":"96"}
```
```
{"at":"2020-02-19T09:11:18.63Z","env":"1.5.0:baf2a","ns":["cardano","cli","generate-txs","submit"],"data":{"time":"2020-02-19 09:11:18.630450099 UTC","kind":"TxSubmissionRecvRequestTxs","txIds":["txid: txid:d0d4058e"]},"app":[],"msg":"","pid":"1146","loc":null,"host":"","sev":"Debug","thread":"96"}
```

**Pull request merged**